zerver/lib/subscription_info.py

```diff
@@ -46,12 +46,13 @@ def get_next_color() -> str:
         return color
 
     subscribed = []
-    for stream in get_web_public_streams_queryset(realm):
+    streams = get_web_public_streams_queryset(realm)
+    setting_groups_dict = get_group_setting_value_dict_for_streams(list(streams))
+
+    for stream in streams:
         # Add Stream fields.
         is_archived = stream.deactivated
-        can_remove_subscribers_group = get_group_setting_value_for_api(
-            stream.can_remove_subscribers_group
-        )
+        can_remove_subscribers_group = setting_groups_dict[stream.can_remove_subscribers_group_id]
```

Comment:
It's a bit surprising to me that this doesn't change any query counts.

Reply:
I don't think we test query count for spectators

Reply:
Ahh, OK. Opened #32409 to track that. Not sure if it's actually accessible to folks new to the project, since most of the work is picking a test to extend.